### PR TITLE
Use shared key for worker auth

### PR DESCRIPTION
**Pull request by chefsalim**
Signed-off-by: Salim Alam <salam@chef.io>

This change allows Builder workers to auth without depending on a github auth token. Instead, it introduces a Builder auth token which is based on a shared key between worker and the API front end. The worker config has an added config entry for the shared key dir - this shared key is the same bldr encryption key that the job server and API nodes already use.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-http-gateway/src/http/middleware.rs [hunk 1]
```diff
@@ -228,6 +245,7 @@ pub fn session_create_github(req: &mut Request, token: String) -> IronResult<Ses
     match github.user(&token) {
         Ok(user) => {
             let mut request = SessionCreate::new();
+            request.set_session_type(SessionType::User);
```
Comment (reset) [inline components/builder-http-gateway/src/http/middleware.rs]: 👍

**Comment by reset** [comment]
tenor-213946436

@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/325576968).

**Comment by raskchanky** [comment]
tenor-8764594

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/325576968).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**